app/services/articles/feeds/large_forem_experimental.rb

```diff
@@ -28,11 +28,10 @@ def find_featured_story(stories)
       end
 
       def published_articles_by_tag
-        articles = Article.published.limited_column_select
+        articles = @tag.present? ? Tag.find_by(name: @tag).articles : Article
```

Comment:
Question: would `Article.tagged_with(@tag)` also work? 

That performs a single SQL query vs the two that this generates:

* `Tag.find_by(name: @tag).articles`:
    ```ruby
  Tag Load (1.3ms)  SELECT "tags".* FROM "tags" WHERE "tags"."name" = $1 LIMIT $2  [["name", "beginners"], ["LIMIT", 1]]
  Article Load (12.9ms)  SELECT "articles".* FROM "articles" INNER JOIN "taggings" ON "articles"."id" = "taggings"."taggable_id" WHERE "taggings"."tag_id" = $1 AND "taggings"."taggable_type" = $2  [["tag_id", 1], ["taggable_type", "Article"]]
    ```
* `Articles.tagged_with(@tag)`:
    ```ruby
    Article Load (2.7ms)  SELECT "articles".* FROM "articles" INNER JOIN "taggings" "article_taggings_ee8d301" ON "article_taggings_ee8d301"."taggable_id" = "articles"."id" AND "article_taggings_ee8d301"."taggable_type" = 'Article' AND "article_taggings_ee8d301"."tag_id" IN (SELECT "tags"."id" FROM "tags" WHERE "tags"."name" LIKE 'beginners' ESCAPE '!')
```

Reply:
@citizen428 the join is actually faster on DEV and it's not sensibly slower on smaller Forems. That's why we're replacing `articles.cahed_tagged_wth(@tag)` with it.

See https://github.com/forem/forem/pull/12581, https://dev.to/admin/blazer/queries/320-query-plan-count-articles-story-controller-1 and https://dev.to/admin/blazer/queries/321-query-plan-count-articles-story-controller-2

@maestromac we should write a forem.team post about this discovery, inspired by @citizen428's latest one on primary keys

Reply:
Just testing on my local instance with <200 articles, the difference between the old and new query is only a few millisecond difference. My hunch is It would be safe to assume the same on newer Forem instances.

>@maestromac we should write a forem.team post about this discovery, inspired by @citizen428's latest one on primary keys

Yeah I can write something up.

That GIN index looks very promising though and could be our next strategy if what we had here slowed down in the future.

Reply:
Thank you @maestromac!

Reply:
@rhymes and @maestromac Thanks for going into all this detail!

My "concern" wasn't only about the generated queries, but also about consistently using the same way to get articles for tags.  Given that we are using `tagged_with` in other contexts too, I wonder if adding the GIN index is just generally a good idea to speed up this type of tag query and we get consistency in the code as a bonus.

Reply:
@citizen428 good point! This way we don't re-structure all the code :)

Reply:
Yes that is a very good point

Reply:
😃 Nice. Would one of you want to pick up creating the index (and maybe changing this one line of code here) as a "15 minute task"? I think we can link the PR to the discussion here.

Reply:
Hey sorry for the late response. Seeing that the speed is fast enough using the existing index with the only downside being inconsistent syntax, wouldn't it be alright to just leave it as is? I can create an issue for a future follow-up so it's not forgotten.

Reply:
I'll leave this up to your judgement @maestromac 😃 I jut thought it might be a good idea to add the index to speed up all `tagged_with` queries across the code base, but maybe that's not necessary.